sympy/functions/elementary/exponential.py

```diff
@@ -460,11 +460,14 @@ def _eval_rewrite_as_sqrt(self, arg):
         from sympy.functions.elementary.trigonometric import sin, cos
         if arg.is_Mul:
             coeff = arg.coeff(S.Pi*S.ImaginaryUnit)
-            cosine, sine = cos(S.Pi*coeff), sin(S.Pi*coeff)
-            if not isinstance(cosine, cos) and not isinstance (sine, sin):
-                return cosine + S.ImaginaryUnit*sine
+            if (coeff == 0):
```

Comment:
This means that `arg` is not a multiple of `pi*I`. In this case `None` should be returned, which means that no rewrite took place.

Reply:
@jksuom what about `exp(x*I).rewrite(sqrt)` because here also coeff=0

Reply:
It cannot be rewritten, so the final result should be `exp(x*I)`. (Return value of 1 is a bug.)